fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java

```diff
@@ -170,163 +193,201 @@ public void testSearchQuantity_Quantity_NoDisplayUnit() throws Exception {
         assertSearchReturnsSavedResource("Range", "ne11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_AP() throws Exception {
         assertSearchDoesntReturnSavedResource("Range", "ap4||s");
         assertSearchReturnsSavedResource("Range", "ap5||s");
         assertSearchReturnsSavedResource("Range", "ap10||s");
         assertSearchDoesntReturnSavedResource("Range", "ap11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_EQ_Implied() throws Exception {
+        // the range of the search value doesn't fully contain the range of the target value
+        assertSearchDoesntReturnSavedResource("Range", "4||s");
+        assertSearchDoesntReturnSavedResource("Range", "5||s");
+        assertSearchReturnsSavedResource("Range", "10||s");
+        assertSearchDoesntReturnSavedResource("Range", "11||s");
+    }
+
+    @Test
+    public void testSearchQuantity_Range_LT() throws Exception {
         assertSearchDoesntReturnSavedResource("Range", "lt4||s");
         assertSearchDoesntReturnSavedResource("Range", "lt5||s");
-        assertSearchReturnsSavedResource("Range", "lt10||s");
+
+        // Lower Bound is 9.5 for 11, therefore adding precision.
+        // And this is still a RANGE search.
+        assertSearchReturnsSavedResource("Range", "lt10.05||s");
         assertSearchReturnsSavedResource("Range", "lt11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_GT() throws Exception {
         assertSearchReturnsSavedResource("Range", "gt4||s");
         assertSearchReturnsSavedResource("Range", "gt5||s");
         assertSearchDoesntReturnSavedResource("Range", "gt10||s");
         assertSearchDoesntReturnSavedResource("Range", "gt11||s");
-        
-        assertSearchDoesntReturnSavedResource("Range", "le4||s");
-        assertSearchReturnsSavedResource("Range", "le5||s");
-        assertSearchReturnsSavedResource("Range", "le10||s");
-        assertSearchReturnsSavedResource("Range", "le11||s");
-        
-        assertSearchReturnsSavedResource("Range", "ge4||s");
-        assertSearchReturnsSavedResource("Range", "ge5||s");
-        assertSearchReturnsSavedResource("Range", "ge10||s");
-        assertSearchDoesntReturnSavedResource("Range", "ge11||s");
-        
-        assertSearchReturnsSavedResource("Range", "sa4||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa5||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa10||s");
-        assertSearchDoesntReturnSavedResource("Range", "sa11||s");
-        
+    }
+
+    @Test
+    public void testSearchQuantity_Range_EB() throws Exception {
         assertSearchDoesntReturnSavedResource("Range", "eb4||s");
         assertSearchDoesntReturnSavedResource("Range", "eb5||s");
-        assertSearchDoesntReturnSavedResource("Range", "eb10||s");
+        // We use the range, so we actually return the value here. 
+        assertSearchReturnsSavedResource("Range", "eb10||s");
```

Comment:
Probably not a big deal, but I disagree with this change.  https://www.hl7.org/fhir/datatypes.html#Range says
> The low and the high values are inclusive and are assumed to have arbitrarily high precision; e.g. the range 1.5 to 2.5 includes 1.50, and 2.50 but not 1.49 or 2.51

Because 5-10 include 10.000... and eb10 includes 10.000 (both with arbitrarilly high precision) I think it should return empty.